pkg/tfvars/openstack/rhcos_image.go

```diff
@@ -0,0 +1,139 @@
+package openstack
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imagedata"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imageimport"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/sirupsen/logrus"
+)
+
+// uploadBaseImage creates a new image in Glance and uploads the RHCOS image there
+func uploadBaseImage(cloud string, localFilePath string, imageName string, clusterID string) error {
+	logrus.Debugln("Creating a Glance image for RHCOS...")
+
+	f, err := os.Open(localFilePath)
+	if err != nil {
+		return err
+	}
+
+	opts := clientconfig.ClientOpts{
+		Cloud: cloud,
+	}
+
+	conn, err := clientconfig.NewServiceClient("image", &opts)
+	if err != nil {
+		return err
+	}
+
+	imageCreateOpts := images.CreateOpts{
+		Name:            imageName,
+		ContainerFormat: "bare",
+		DiskFormat:      "raw",
+		Tags:            []string{fmt.Sprintf("openshiftClusterID=%s", clusterID)},
+		// TODO(mfedosin): add Description when gophercloud supports it.
+	}
+
+	img, err := images.Create(conn, imageCreateOpts).Extract()
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("Image %s was created.", img.Name)
+
+	useImageImport, err := isImageImportSupported(cloud)
+	if err != nil {
+		return err
+	}
+
+	if useImageImport {
```

Comment:
Could you find a cloud with glance direct image import to test this?

Reply:
I tested it on manila-env-like devstack with Glance and Ceph... This is not really cool, I know. We'll need to obtain a real OSP13 env, where we can test the solution properly.